cmd/soroban-rpc/internal/ingest/service.go

```diff
@@ -208,9 +209,14 @@ func (s *Service) fillEntriesFromCheckpoint(ctx context.Context, archive history
 	if err != nil {
 		return err
 	}
+	transactionCommitted := false
 	defer func() {
-		if err := tx.Rollback(); err != nil {
-			s.logger.WithError(err).Warn("could not rollback fillEntriesFromCheckpoint write transactions")
+		if transactionCommitted == false {
+			// Internally, we might already have rolled back the transaction. We should
+			// not generate benign error/warning here in case the transaction was already rolled back.
+			if err := tx.Rollback(); err != nil && err != supportdb.ErrAlreadyRolledback {
```

Comment:
golangci - Consider renaming this so it doesn't shadow the declaration of `err` above

Reply:
yup.